pynsee/utils/save_df.py

```diff
@@ -0,0 +1,124 @@
+import functools
+import os
+import pandas as pd
+import warnings
+import shapely
+from shapely.errors import ShapelyDeprecationWarning
+import warnings
+import datetime 
+import logging
+logger = logging.getLogger(__name__)
+
+from pynsee.utils._create_insee_folder import _create_insee_folder
+from pynsee.utils._hash import _hash
+
+@functools.lru_cache(maxsize=None)
+def _warning_cached_data(file, mdate=None, day_lapse=None):
+    strg_print = f"Previously saved data has been used:\n{file}\n"
+    if (mdate is not None) and (day_lapse is not None):
+        strg_print += f"Creation date: {mdate:%Y-%m-%d}"
+        
+        if day_lapse >= 2:
+            strg_print += ", {day_lapse} days ago\n"
+        if day_lapse == 1:
+            strg_print += ", yesterday\n"
+        if day_lapse == 0:
+            strg_print += ", today\n"
+            
+    strg_print += "Set update=True to get the most up-to-date data"
+
+    logger.info(strg_print)
+
+def save_df(obj=pd.DataFrame, parquet=True, day_lapse_max=None):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            
+            data_folder = _create_insee_folder()
+            string_file_arg = [str(kwargs[a]) for a in kwargs.keys() if a not in ['update', 'silent']]
+            string_file_arg += [func.__name__] + [str(a) for a in args]            
+            file_name = data_folder + "/" + _hash(''.join(string_file_arg)) 
+            
+            if parquet:
+                file_name += ".parquet"
+            else:
+                file_name += ".pkl"            
+                                    
+            update = any((a == 'update') and (kwargs[a] is True) for a in kwargs)
+
+            silent = any((a == 'silent') and (kwargs[a] is True) for a in kwargs)
+            
+            if os.path.exists(file_name):
+                file_date_last_modif = datetime.datetime.fromtimestamp(
+                    os.path.getmtime(file_name)
+                )
+                try:
+                    # only used for testing purposes
+                    insee_date_time_now = os.environ["insee_date_test"]
+                    insee_date_time_now = datetime.datetime.strptime(
+                        insee_date_time_now, "%Y-%m-%d %H:%M:%S.%f"
+                    )
```

Comment:
what is this?
if it's for testing only, shouldn't it check ``kwargs`` for a testing value or something, this looks quite strange...

Reply:
the following commit uses kwarg  as advised to test the function retriggering feature if cached data is too old
607d370f9d71764561f8e516803a1ae096aa565f